### PR TITLE
Install logs: NoWorkerNodes => (AWS|GCP)NoWorkerNodes

### PR DESCRIPTION
**Pull request by 2uasimojo**
Replace the `NoWorkerNodes` regex/reason with cloud-specific versions
`AWSNoWorkerNodes` and `GCPNoWorkerNodes`.

[HIVE-1722](https://issues.redhat.com/browse/HIVE-1722)

### CODE REVIEW

**Comment by 2uasimojo** [comment]
Are we going to have any issues with this _removing_ the old failure reason code?

**Comment by joelddiaz** [comment]
/hold
in fact, now that i see that these regex configmaps are more than just OSD-specific (eg ACM would also receive this), i'd really like to see more generic text associated with this

File: config/configmaps/install-log-regexes-configmap.yaml [hunk 1]
```diff
@@ -163,6 +164,12 @@ data:
       - "iam\\.googleapis\\.com/quota/service-account-count is not available in global because the required number of resources \\([0-9]*\\) is more than remaining quota"
       installFailingReason: GCPServiceAccountQuotaExceeded
       installFailingMessage: GCP Service Account quota exceeded
+    # cf. AWSNoWorkerNodes
+    - name: GCPNoWorkerNodes
+      searchRegexStrings:
+      - "(?s)terraform-provider-gcp.*Got 0 worker nodes, 3 master nodes"
+      installFailingReason: GCPNoWorkerNodes
+      installFailingMessage: No worker nodes could be created. Check that your machine-api role is correct and try again.
```
[the remark this answers is not in the document]
Comment (2uasimojo) [reply config/configmaps/install-log-regexes-configmap.yaml:172]: Yeah, no idea -- I was just using what was there before. Happy to take suggestions for a more appropriate message.

[the remark this answers is not in the document]
Comment (joelddiaz) [reply config/configmaps/install-log-regexes-configmap.yaml:172]: In the absence of more info explaining why this definitely should be assumed to be a permissions issue, I would back it off to something like "worker nodes never appeared, check machine-api logs"

[the remark this answers is not in the document]
Comment (2uasimojo) [reply config/configmaps/install-log-regexes-configmap.yaml:172]: Can do. Would you also like me to soften the `3 master nodes` bit? E.g. `\d+ master nodes`?

[the remark this answers is not in the document]
Comment (joelddiaz) [reply config/configmaps/install-log-regexes-configmap.yaml:172]: i don't know enough about the intricacies of the installation routine on whether we could ever get this far in the install without having first successfully brought up 3 master nodes ¯\_(ツ)_/¯

[the remark this answers is not in the document]
Comment (2uasimojo) [reply config/configmaps/install-log-regexes-configmap.yaml]: Can't hurt, really. Done.

File: config/configmaps/install-log-regexes-configmap.yaml [hunk 2]
```diff
@@ -88,11 +88,12 @@ data:
...
-    - name: NoWorkerNodes
+    # cf. GCPNoWorkerNodes
+    - name: AWSNoWorkerNodes
       searchRegexStrings:
-      - "Got 0 worker nodes, 3 master nodes"
-      installFailingReason: NoWorkerNodes
-      installFailingMessage: No worker nodes could be created. Check that your machine-api role is correct and try again.
+      - "(?s)terraform-provider-aws.*Got 0 worker nodes, \\d+ master nodes"
+      installFailingReason: AWSNoWorkerNodes
+      installFailingMessage: No worker nodes could be created. Check the machine-api logs.
```
Comment (gregsheremeta) [inline config/configmaps/install-log-regexes-configmap.yaml:96]:
```suggestion
      installFailingMessage: No worker nodes could be created
```

File: config/configmaps/install-log-regexes-configmap.yaml [hunk 3]
```diff
@@ -163,6 +164,12 @@ data:
       - "iam\\.googleapis\\.com/quota/service-account-count is not available in global because the required number of resources \\([0-9]*\\) is more than remaining quota"
       installFailingReason: GCPServiceAccountQuotaExceeded
       installFailingMessage: GCP Service Account quota exceeded
+    # cf. AWSNoWorkerNodes
+    - name: GCPNoWorkerNodes
+      searchRegexStrings:
+      - "(?s)terraform-provider-gcp.*Got 0 worker nodes, \\d+ master nodes"
+      installFailingReason: GCPNoWorkerNodes
+      installFailingMessage: No worker nodes could be created. Check the machine-api logs.
```
Comment (gregsheremeta) [inline config/configmaps/install-log-regexes-configmap.yaml:172]:
```suggestion
      installFailingMessage: No worker nodes could be created
```

File: pkg/operator/assets/bindata.go [hunk 4]
```diff
@@ -1650,11 +1650,12 @@ data:
...
-    - name: NoWorkerNodes
+    # cf. GCPNoWorkerNodes
+    - name: AWSNoWorkerNodes
       searchRegexStrings:
-      - "Got 0 worker nodes, 3 master nodes"
-      installFailingReason: NoWorkerNodes
-      installFailingMessage: No worker nodes could be created. Check that your machine-api role is correct and try again.
+      - "(?s)terraform-provider-aws.*Got 0 worker nodes, \\d+ master nodes"
+      installFailingReason: AWSNoWorkerNodes
+      installFailingMessage: No worker nodes could be created. Check the machine-api logs.
```
Comment (gregsheremeta) [inline pkg/operator/assets/bindata.go:1658]:
```suggestion
      installFailingMessage: No worker nodes could be created
```

File: pkg/operator/assets/bindata.go [hunk 5]
```diff
@@ -1725,6 +1726,12 @@ data:
       - "iam\\.googleapis\\.com/quota/service-account-count is not available in global because the required number of resources \\([0-9]*\\) is more than remaining quota"
       installFailingReason: GCPServiceAccountQuotaExceeded
       installFailingMessage: GCP Service Account quota exceeded
+    # cf. AWSNoWorkerNodes
+    - name: GCPNoWorkerNodes
+      searchRegexStrings:
+      - "(?s)terraform-provider-gcp.*Got 0 worker nodes, \\d+ master nodes"
+      installFailingReason: GCPNoWorkerNodes
+      installFailingMessage: No worker nodes could be created. Check the machine-api logs.
```
Comment (gregsheremeta) [inline pkg/operator/assets/bindata.go:1734]:
```suggestion
      installFailingMessage: No worker nodes could be created
```

**Review by gregsheremeta — COMMENTED** [review]
> In the absence of more info explaining why this definitely should be assumed to be a permissions issue, I would back it off to something like "worker nodes never appeared, check machine-api logs"

My suggestions removes that second part. We don't use this message in OSD (we craft a custom message in OCM) so plain ol' "no worker nodes could be created" is absolutely fine.

**Review by joelddiaz — COMMENTED** [review]
/lgtm

**Pull request merged**